src/renderer/coremods/settings/pages/General.tsx

```diff
@@ -1,23 +1,116 @@
+import { modal } from "@common";
 import React from "@common/react";
+import { Button, Divider, Flex, FormItem, SwitchItem, Text, TextInput } from "@components";
+import * as settings from "../../../apis/settings";
+import * as util from "../../../util";
+import { Messages } from "@common/i18n";
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type GeneralSettings = {
+  apiUrl: string;
+  // pluginEmbeds: boolean;
+  experiments: boolean;
+};
+
+export const defaultSettings: Partial<GeneralSettings> = {
+  apiUrl: "https://replugged.dev",
+  // pluginEmbeds: false,
+  experiments: false,
+};
+
+export const generalSettings = await settings.init<GeneralSettings, keyof typeof defaultSettings>(
+  "rp-settings",
+  defaultSettings,
+);
 
 export const General = (): React.ReactElement => {
+  const { value: expValue, onChange: expOnChange } = util.useSetting(
+    generalSettings,
+    "experiments",
+    false,
+  );
   const [hue, setHue] = React.useState(0);
+  const [sleep, setSleep] = React.useState(false);
   React.useEffect(() => {
     const id = requestAnimationFrame(() => {
-      // ESLint thinks that hue is not a number for some reason.
-
       setHue((hue + 1) % 360);
     });
     return () => cancelAnimationFrame(id);
   }, [hue]);
 
+  /* @see view-source:https://sysspa.alyxia.dev/ lines 172 to 183 */
+  const konami = "38,38,40,40,37,39,37,39,66,65";
+  const kKeys: number[] = [];
+  let listener: (e: KeyboardEvent) => void;
+  document.addEventListener(
+    "keydown",
+    (listener = function (e) {
+      kKeys.push(e.keyCode);
+
+      if (kKeys.toString().includes(konami)) {
+        document.removeEventListener("keydown", listener);
+
+        setSleep(true);
+      }
+    }),
+  );
```

Comment:
This is adding a new event listener on every render. Let's put this in a useEffect with a cleanup function to remove the listener when this is unmounted.

Reply:
I don't know how to do that.